sympy/integrals/tests/test_transforms.py

```diff
@@ -665,56 +666,70 @@ def test_laplace_transform():
     # Test general rules and unevaluated forms
     # These all also test whether issue #7219 is solved.
     assert LT(Heaviside(t-1)*cos(t-1), t, s) == (s*exp(-s)/(s**2 + 1), 0, True)
-    assert LT(a*f(t), t, w) == a*LaplaceTransform(f(t), t, w)
+    assert LT(a*f(t), t, w) == (a*LaplaceTransform(f(t), t, w), -oo, True)
```

Comment:
I guess this is an example where conditions were not previously returned. We don't have conditions because we don't know when the Laplace transform of `f(t)` converges but substituting a function could lead to convergence conditions:
```python
In [1]: a, b, t, s = symbols('a, b, t, s')

In [2]: laplace_transform(a*f(t), t, s)
Out[2]: (a⋅LaplaceTransform(f(t), t, s), -∞, True)

In [3]: laplace_transform(a*f(t), t, s)[0].subs(f, exp)
Out[3]: 
                  ⎛ t      ⎞
a⋅LaplaceTransform⎝ℯ , t, s⎠

In [4]: laplace_transform(a*f(t), t, s)[0].subs(f, exp).doit()
/home/oscar/current/sympy/sympy.git/sympy/core/operations.py:458: SymPyDeprecationWarning: 

Using non-Expr arguments in Mul is deprecated (in this case, one of
the arguments has type 'Tuple').

If you really did intend to use a multiplication or addition operation with
this object, use the * or + operator instead.

See https://docs.sympy.org/latest/explanation/active-deprecations.html#non-expr-args-deprecated
for details.

This has been deprecated since SymPy version 1.7. It
will be removed in a future version of SymPy.

  return self.func(*terms, evaluate=True)
Out[4]: 
  ⎛  1           ⎞
a⋅⎜─────, 1, True⎟
  ⎝s - 1         ⎠
```
The problem here is that calling doit on an unevaluated LaplaceTransform results in a tuple which cannot really be substituted for a LaplaceTransform as part of an expression.

I think that if LaplaceTransform is supposed to be an Expr then LaplaceTransform.doit() should give an Expr which means that it should not give convergence conditions. Only `laplace_transform` should do that...

Reply:
Ah ... I can do that by giving `LaplaceTransform` the option `noconds=True` as the standard setting and call it with `.doit(noconds=False)` from within `laplace_transform`. Would that be an acceptable solution? It would definitely solve the problem with your example.

Reply:
> Would that be an acceptable solution?

Maybe. Would that be a breaking change?

In general I think that `doit` should not return a different kind of object regardless of any hints. Also it should always return a symbolic object i.e. a Basic instance and in particular for any Expr we should have that Expr.doit also gives an Expr. Returning a tuple from doit is generally a broken feature.

This also related to my point that recursing through LaplaceTransform.doit just doesn't seem right to me. There should be a different function or method through which recursion happens so that we are clear what the purpose of doit itself is. It's important that doit be usable in a different kind of recursion which is that an expression rebuilds itself by recursively applying doit through the args of each subexpression.

Reply:
That one was surprisingly easy ...

Reply:
> Maybe. Would that be a breaking change?

No, it changes nothing at all from the outside.

Reply:
Except, of course, that recursively applying .doit() on subexpressions now does what you want ;)